alg/gdal_alg.h

```diff
@@ -353,8 +353,8 @@ GDALDatasetH CPL_DLL GDALViewshedGenerate(
     double dfTargetHeight, double dfVisibleVal, double dfInvisibleVal,
     double dfOutOfRangeVal, double dfNoDataVal, double dfCurvCoeff,
     GDALViewshedMode eMode, double dfMaxDistance, GDALProgressFunc pfnProgress,
-    void *pProgressArg, GDALViewshedOutputType heightMode,
-    CSLConstList papszExtraOptions);
+    void *pProgressArg, GDALViewshedOutputType heightMode, double startAngle,
```

Comment:
we cannot change the prototype of existing C functions. Please use the existing papszExtraOptions member to pass them as strings, like

papszExtraOptions = CSLAddNameString(papszExtraOptions, "START_ANGLE", "45");

Reply:
the SWIG API can be extended with "natural parameters" though  (converting the numeric value into strings as shown above)